pkg/rpc/server_helper_test.go

```diff
@@ -0,0 +1,182 @@
+		Time              int    `json:"time"`
+		Height            int    `json:"height"`
+		Nonce             int    `json:"nonce"`
+		NextConsensus     string `json:"next_consensus"`
+		Script            struct {
+			Invocation   string `json:"invocation"`
+			Verification string `json:"verification"`
+		} `json:"script"`
+		Tx []struct {
+			Type       string      `json:"type"`
+			Version    int         `json:"version"`
+			Attributes interface{} `json:"attributes"`
+			Vin        interface{} `json:"vin"`
+			Vout       interface{} `json:"vout"`
+			Scripts    interface{} `json:"scripts"`
+		} `json:"tx"`
+		Confirmations int    `json:"confirmations"`
+		Nextblockhash string `json:"nextblockhash"`
+		Hash          string `json:"hash"`
+	} `json:"result"`
+	ID int `json:"id"`
+}
+
+// GetAssetResponse struct for testing.
+type GetAssetResponse struct {
+	Jsonrpc string `json:"jsonrpc"`
+	Result  struct {
+		AssetID    string `json:"assetID"`
+		AssetType  int    `json:"assetType"`
+		Name       string `json:"name"`
+		Amount     string `json:"amount"`
+		Available  string `json:"available"`
+		Precision  int    `json:"precision"`
+		Fee        int    `json:"fee"`
+		Address    string `json:"address"`
+		Owner      string `json:"owner"`
+		Admin      string `json:"admin"`
+		Issuer     string `json:"issuer"`
+		Expiration int    `json:"expiration"`
+		IsFrozen   bool   `json:"is_frozen"`
+	} `json:"result"`
+	ID int `json:"id"`
+}
+
+func initServerWithInMemoryChain(ctx context.Context, t *testing.T) (*core.Blockchain, http.HandlerFunc) {
+	net := config.ModeUnitTestNet
+	configPath := "../../config"
+	cfg, err := config.Load(configPath, net)
+	require.NoError(t, err, "could not load config")
+
+	memoryStore := storage.NewMemoryStore()
+	chain, err := core.NewBlockchain(memoryStore, cfg.ProtocolConfiguration)
+	require.NoError(t, err, "could not create chain")
+
+	go chain.Run(ctx)
+	initBlocks(t, chain)
+
+	serverConfig := network.NewServerConfig(cfg)
+	server := network.NewServer(serverConfig, chain)
+	rpcServer := NewServer(chain, cfg.ApplicationConfiguration.RPCPort, server)
+	handler := http.HandlerFunc(rpcServer.requestHandler)
+
+	return chain, handler
+}
+
+func initBlocks(t *testing.T, chain *core.Blockchain) {
+	blocks := makeBlocks(10)
+	for i := 0; i < len(blocks); i++ {
+		require.NoError(t, chain.AddBlock(blocks[i]))
+	}
+	require.NoError(t, chain.Persist(context.Background()))
+}
+
+func makeBlocks(n int) []*core.Block {
+	blocks := make([]*core.Block, n)
+	for i := 0; i < n; i++ {
+		blocks[i] = newBlock(uint32(i+1), newTX(transaction.MinerType))
+	}
+	return blocks
+}
+
+func newTX(t transaction.TXType) *transaction.Transaction {
+	return &transaction.Transaction{
+		Type: t,
+	}
+}
+
+func newBlock(index uint32, txs ...*transaction.Transaction) *core.Block {
+	b := &core.Block{
+		BlockBase: core.BlockBase{
+			Version:       0,
+			PrevHash:      hash.Sha256([]byte("a")),
+			MerkleRoot:    hash.Sha256([]byte("b")),
```

Comment:
I think this will break when we're to add proper block verifications.

Reply:
Totally agree with that. Even other tests in block_test.go will fail as well since they use the same code. I think this part has to be reworked in separate patchset since it will touch a lot of places and probably we will throw these tests to Integration tests package and will do block creation on other way.